selectors/views/threads.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {createSelector} from 'reselect';
+
+import {$ID} from 'mattermost-redux/types/utilities';
+import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
+import {getThreads} from 'mattermost-redux/selectors/entities/threads';
+import {Team} from 'mattermost-redux/types/teams';
+import {UserThread} from 'mattermost-redux/types/threads';
+
+import {GlobalState} from 'types/store';
+import {ViewsState} from 'types/store/views';
+
+export function getSelectedThreadIdInTeam(state: GlobalState) {
+    return state.views.threads.selectedThreadIdInTeam;
+}
+
+export const getSelectedThreadIdInCurrentTeam: (state: GlobalState) => ViewsState['threads']['selectedThreadIdInTeam'][$ID<Team>] = createSelector(
+    getCurrentTeamId,
+    getSelectedThreadIdInTeam,
+    (
+        currentTeamId,
+        selectedThreadIdInTeam,
+    ) => {
+        return selectedThreadIdInTeam?.[currentTeamId] ?? null;
+    },
+);
+
+export const getSelectedThreadInCurrentTeam: (state: GlobalState) => UserThread | null = createSelector(
```

Comment:
Same question as above re: `createSelector`.

Reply:
Generally, I haven't done extensive performance testing on the lower end here with these smaller selectors, so I'm not sure if they need memoization, but I generally default to them akin to preferring `PureComponent` over `Component`.

At this late stage, I'm tempted to say that we should leave them as-is for now, and address any no-need-to-memoize issues in a follow up PR shortly afterwards that would address any performance learnings (on both the too slow/not enough and no-need ends).